Source/Extensions/Blazorise.TreeView/Internal/_TreeViewNodeContent.razor.cs

```diff
@@ -1,113 +1,146 @@
-    }
-
-    protected override Task OnParametersSetAsync()
-    {
-        if ( Selected )
-            SelectedNodeStyling?.Invoke( Node, selectedNodeStyling );
-        else
-            NodeStyling?.Invoke( Node, nodeStyling );
-
-        return base.OnParametersSetAsync();
-    }
-
-    #endregion
-
-    #region Properties
-
-    protected bool Selected
-        => TreeViewState.SelectedNode != null && TreeViewState.SelectedNode.Equals( Node );
-
-    [Parameter] public TNode Node { get; set; }
-
-    [CascadingParameter] public TreeView<TNode> Parent { get; set; }
-
-    [CascadingParameter]
-    protected TreeViewState<TNode> TreeViewState
-    {
-        get => treeViewState;
-        set
-        {
-            if ( treeViewState == value )
-                return;
-
-            treeViewState = value;
-
-            DirtyClasses();
-        }
-    }
-
-    [Parameter] public RenderFragment ChildContent { get; set; }
-
-    /// <summary>
-    /// Gets or sets selected node styling.
-    /// </summary>
-    [Parameter] public Action<TNode, NodeStyling> SelectedNodeStyling { get; set; }
-
-    /// <summary>
-    /// Gets or sets node styling.
-    /// </summary>
-    [Parameter] public Action<TNode, NodeStyling> NodeStyling { get; set; }
-
-    #endregion
+﻿#region Using directives
+using System;
+using System.Threading.Tasks;
+using Blazorise.Utilities;
+using Microsoft.AspNetCore.Components;
+#endregion
+
+namespace Blazorise.TreeView.Internal;
+
+public partial class _TreeViewNodeContent<TNode> : BaseComponent
+{
+    #region Members
+
+    private TreeViewState<TNode> treeViewState;
+
+    private NodeStyling selectedNodeStyling;
+
+    private NodeStyling nodeStyling;
+
+    private TreeViewSelectionMode selectionMode;
+
+    #endregion
+
+    #region Constructors
+
+    public _TreeViewNodeContent()
+    {
+        selectedNodeStyling = new()
+        {
+            Background = Background.Primary,
+            TextColor = TextColor.White
+        };
+
+        nodeStyling = new()
+        {
+            Background = Background.Default,
+            TextColor = TextColor.Default
+        };
+    }
+
+    #endregion
+
+    #region Methods
+
+    protected override void BuildClasses( ClassBuilder builder )
+    {
+        builder.Append( $"{ClassProvider.Spacing( Spacing.Padding, SpacingSize.Is1, Side.All, Breakpoint.None )} cursor-pointer" );
+
+        if ( Selected )
+            builder.Append( $"{ClassProvider.BackgroundColor( selectedNodeStyling.Background )} {ClassProvider.TextColor( selectedNodeStyling.TextColor )} {selectedNodeStyling.Class}" );
+        else
+            builder.Append( $"{ClassProvider.BackgroundColor( nodeStyling.Background )} {ClassProvider.TextColor( nodeStyling.TextColor )} {nodeStyling.Class}" );
+
+        base.BuildClasses( builder );
+    }
+
+    protected Task OnClick()
+    {
+        if ( SelectionMode != TreeViewSelectionMode.Single )
+            return Task.CompletedTask;
```

Comment:
Are we disabling regular selection when multiple? I think people still like that the rows can be selected either on checkbox or row?

Reply:
Well, it comes to my mind. But maybe later on, people would want to click on a node without making it checked. So I left it only to be selectable with a checkbox.